fe/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -497,4 +596,108 @@ public static OlapScanNode createOlapScanNodeByLocation(
 
         return olapScanNode;
     }
+
+    public void collectColumns(Analyzer analyzer, Set<String> equivalenceColumns, Set<String> unequivalenceColumns) {
+        // 1. Get columns which has predicate on it.
+        for (Expr expr : conjuncts) {
+            if (!isPredicateUsedForPrefixIndex(expr, false)) {
+                continue;
+            }
+            for (SlotDescriptor slot : desc.getMaterializedSlots()) {
+                if (expr.isBound(slot.getId())) {
+                    if (!isEquivalenceExpr(expr)) {
+                        unequivalenceColumns.add(slot.getColumn().getName());
+                    } else {
+                        equivalenceColumns.add(slot.getColumn().getName());
+                    }
+                    break;
+                }
+            }
+        }
+
+        // 2. Equal join predicates when pushing inner child.
+        List<Expr> eqJoinPredicate = analyzer.getEqJoinConjuncts(desc.getId());
+        for (Expr expr : eqJoinPredicate) {
+            if (!isPredicateUsedForPrefixIndex(expr, true)) {
+                continue;
+            }
+            for (SlotDescriptor slot : desc.getMaterializedSlots()) {
+                Preconditions.checkState(expr.getChildren().size() == 2);
+                for (Expr child : expr.getChildren()) {
+                    if (child.isBound(slot.getId())) {
+                        equivalenceColumns.add(slot.getColumn().getName());
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+
```

Comment:
I think we would better move following function to Expr related class, and change the method name more general for `xxxForPrefixIndex` methods. Because I think these function is useful when we implement new Cascades query optimizer.

Reply:
Why not move them?